pkg/kv/kvserver/client_raft_test.go

```diff
@@ -4057,61 +4057,52 @@ func TestTransferRaftLeadership(t *testing.T) {
 	defer log.Scope(t).Close(t)
 
 	ctx := context.Background()
-	tc := testcluster.StartTestCluster(t, 2,
-		base.TestClusterArgs{
-			ReplicationMode: base.ReplicationManual,
-			ServerArgs: base.TestServerArgs{
-				RaftConfig: base.RaftConfig{
-					// Suppress timeout-based elections (which also includes a previous
-					// leader stepping down due to a quorum check). Running tests on a
-					// heavily loaded CPU is enough to reach the raft election timeout
-					// and cause leadership to change hands in ways this test doesn't
-					// expect.
-					RaftElectionTimeoutTicks: 100000,
-				},
+	tc := testcluster.StartTestCluster(t, 2, base.TestClusterArgs{
+		ReplicationMode: base.ReplicationManual,
+		ServerArgs: base.TestServerArgs{
+			RaftConfig: base.RaftConfig{
+				// Suppress timeout-based elections (which also includes a previous
+				// leader stepping down due to a quorum check). Running tests on a
+				// heavily loaded CPU is enough to reach the raft election timeout and
+				// cause leadership to change hands in ways this test doesn't expect.
+				RaftElectionTimeoutTicks: 100000,
 			},
-		})
+		},
+	})
 	defer tc.Stopper().Stop(ctx)
 	store0 := tc.GetFirstStoreFromServer(t, 0)
 	store1 := tc.GetFirstStoreFromServer(t, 1)
 
 	key := tc.ScratchRangeWithExpirationLease(t)
 	repl0 := store0.LookupReplica(keys.MustAddr(key))
-	if repl0 == nil {
-		t.Fatalf("no replica found for key '%s'", key)
-	}
+	require.NotNil(t, repl0, "no replica found for key '%s'", key)
 	rd0, err := repl0.GetReplicaDescriptor()
-	if err != nil {
-		t.Fatal(err)
-	}
+	require.NoError(t, err)
+
 	tc.AddVotersOrFatal(t, key, tc.Target(1))
 
+	// NB: if we don't wait until node 2 applies the config change and becomes
+	// voter, it may refuse to campaign for leadership. The large Raft election
+	// timeout set in this test will prevent the current leader from retrying the
+	// transfer. See issue #99213.
+	require.NoError(t, tc.WaitForVoters(key, tc.Target(1)))
+
 	repl1 := store1.LookupReplica(keys.MustAddr(key))
-	if repl1 == nil {
-		t.Fatalf("no replica found for key '%s'", key)
-	}
+	require.NotNil(t, repl1, "no replica found for key '%s'", key)
 	rd1, err := repl1.GetReplicaDescriptor()
-	if err != nil {
-		t.Fatal(err)
-	}
+	require.NoError(t, err)
+	require.Equal(t, roachpb.VOTER_FULL, rd1.Type)
 
-	getArgs := getArgs(key)
-	if _, pErr := kv.SendWrappedWith(
-		context.Background(), store0, kvpb.Header{RangeID: repl0.RangeID}, getArgs,
-	); pErr != nil {
-		t.Fatalf("expect get nil, actual get %v ", pErr)
-	}
+	_, pErr := kv.SendWrappedWith(ctx, store0, kvpb.Header{RangeID: repl0.RangeID}, getArgs(key))
+	require.NoError(t, pErr.GoError())
 
-	status := repl0.RaftStatus()
-	if status == nil || status.Lead != uint64(rd0.ReplicaID) {
-		t.Fatalf("raft leader should be %d, but got status %+v", rd0.ReplicaID, status)
-	}
+	require.Equal(t, uint64(rd0.ReplicaID), repl0.RaftStatus().Lead)
```

Comment:
`RaftStatus()` can - in principle - return `nil`, it would be better if this didn't crash the entire test shard.

Reply:
Fixed.